### PR TITLE
Fix changed stddev0 warning

### PR DESCRIPTION
**Pull request by newville**
#### Description

This addresses #999 for the recent changes in how `uncertainties` warns about `ufloat()` values with `std_dev=0`. 

The changes here are:
1.  do not test for the content of warning messages from upstream libraries.  They can change.
2. capture and ignore warnings about `stderr=0` when creating `uncertainties.ufloat()` in `conf_interval()`.

###### Type of Changes

- [x] Bug fix
- [ ] New feature
- [ ] Refactoring / maintenance
- [ ] Documentation / examples

###### Tested on

Python: 3.12.8 | packaged by conda-forge | (main, Dec  5 2024, 14:25:12) [Clang 18.1.8 ]

lmfit: 1.3.3.post3+g132fef4e, scipy: 1.15.1, numpy: 1.26.4, asteval: 1.0.6, uncertainties: 3.2.3

###### Verification 
Have you

- [ ] included docstrings that follow PEP 257?

- [ ] referenced existing Issue and/or provided relevant link to mailing list?

- [ ] verified that existing tests pass locally?

- [ ] verified that the documentation builds locally?

- [ ] squashed/minimized your commits and written descriptive commit messages?

- [x] added or updated existing tests to cover the changes?
- [ ] updated the documentation and/or added an entry to the release notes (doc/whatsnew.rst)?
- [ ] added an example?

### CODE REVIEW

File: lmfit/parameter.py [hunk 1]
```diff
@@ -572,8 +573,13 @@ def create_uvars(self, covar=None):
                     par.stderr = sqrt(covar[vindex, vindex])
             stderr = getattr(par, 'stderr', 0.0)
             if stderr is None:
-                stderr = 0.0
-            uvars[par.name] = ufloat(par.value, stderr)
+                stderr = 0.00
+            if stderr < 1.e-13*abs(par.value):
```
Comment (wshanks) [inline lmfit/parameter.py:577]: `par.value == 0` might be an edge case to handle? I wonder if you could just always filter the warnings here without testing stderr or only test for `stderr==0` instead of with the inequality?

**Review by wshanks — COMMENTED** [review]
I was checking in here since this new warning started causing some tests I work with to fail because they use `vary=False` and are set to fail if any warnings are generated, so I thought I would leave a review.

File: lmfit/parameter.py [hunk 2]
```diff
@@ -590,6 +596,7 @@ def create_uvars(self, covar=None):
             # parameters and reset the Parameters to best-fit value
             wrap_ueval = uwrap(asteval_with_uncertainties)
             for par in self.values():
+                print("xPAR ", par)
```
[the remark this answers is not in the document]
Comment (newville) [reply lmfit/parameter.py:599]: @wshanks Thanks!

File: lmfit/parameter.py (hunk 1 above)
Comment (newville) [reply lmfit/parameter.py:577]: Thanks, that's a good suggestion.  I also see that `tiny` is defined here (as 1.e-15).  
I will change to `if stderr < tiny*max(tiny, abs(par.value)):`, just to be super-paranoid about floating point.

**Review by wshanks — APPROVED** [review]

**Comment by newville** [comment]
Merging...
